pkg/controller/kogitoinfra/kogitoinfra_controller.go

```diff
@@ -107,6 +107,12 @@ func (r *ReconcileKogitoInfra) Reconcile(request reconcile.Request) (result reco
 		return reconcile.Result{}, resultErr
 	}
 
+	// make KogitoInfra as self owner so that it will not removed when kogito service referring to it deleted because
+	// kogito services are also become owner of kogitoInfra when reference of infra provided in Kogito services.
+	if resultErr = framework.AddOwnerReference(instance, r.scheme, instance); resultErr != nil {
```

Comment:
We could take a look on Finalizers as well:
https://sdk.operatorframework.io/docs/building-operators/golang/advanced-topics/#handle-cleanup-on-deletion

Might help in this use case to avoid KogitoInfra being deleted. 

Also, the KogitoInfra owner should not have `controller :true`, this way `KogitoInfra` won't be deleted. This is the approach we are taking when sharing the same ImageStream between `KogitoBuild` and `KogitoRuntime`

Reply:
Yes the owner of kogitoInfra have `controller: false`